src/server/hset_family.cc

```diff
@@ -1060,27 +1061,52 @@ using CI = CommandId;
 
 #define HFUNC(x) SetHandler(&HSetFamily::x)
 
+namespace {
+namespace Acl {
+namespace Cat = AclCategory;
+constexpr uint32_t kHDel = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHLen = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHExists = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHGet = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHGetAll = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHMGet = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHMSet = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHIncrBy = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHIncrByFloat = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHKeys = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHRandField = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHScan = Cat::READ | Cat::HASH | Cat::SLOW;
+constexpr uint32_t kHSet = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHSetEx = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHSetNx = Cat::WRITE | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHStrLen = Cat::READ | Cat::HASH | Cat::FAST;
+constexpr uint32_t kHVals = Cat::READ | Cat::HASH | Cat::SLOW;
+}  // namespace Acl
+}  // namespace
```

Comment:
This is a lot of boilerplate that can actually be inferred from the command flags itself. 
1. All commands contain Cat::HASH
2. CO::Write -> Cat::WRITE, otherwise Cat::READ, CO::FAST -> Cat::FAST, etc..
3. There are rare exclusions like Cat::DANGEROUS that have to be set manually

Reply:
If I was you I'd take the change to play with constexprs a little 😄

Reply:
Really nice observation however I got some concerns:

1. The pattern you observed does not apply to all commands. A good example would be `geoadd` which it has `CO::FAST` but its respective `category` is `Cat::SLOW` and I am pretty sure this is the case for other commands as well. (Although I am not 100% sure if `CO::FAST` is correct here [see)](https://github.com/redis/redis/blob/unstable/src/commands/geoadd.json and I didn't bother to check each one in detail as it would take a lot of time)
2. Following your (2) point: I would be hesitant to introduce any mapping reliance in the constructor since these mappings might change in the future as we reiterate our model. Also I don't know for each command I introduced if there is an 1-1 mapping as I mentioned earlier in (1). 
3. I could easily introduce a constexpr function (instead of doing it in the constructor as in the point above) but that would still require initializing these variables. Yes we might save a few columns, but at the end the difference is not that big.
4. These variables might be used later, so they might be moved in a header. I think it's valuable to be explicit such that it's easy to make changes -- a mapping that breaks our model would maybe be a pain to deal with.

> If I was you I'd take the change to play with constexprs a little 😄

You sell this so nicely that I am even tempted to do this purely from the comment (and I would love to do it, don't get me wrong -- it's just that these mappings can backfire). if . Evil :man_cook: 

Once we have a prototype working and stable, then it would make sense to refactor some parts of this + look on the `CO` staff.

What do you think? 

p.s. sorry for the long comment, my mind exploded from introducing all of these constexpr variables and looking on acl categories all day, but I tried to be as less intrusive as possible such that we can make changes fast in the future.

Reply:
1. How come? I see its marked fast by us and slow by redis. If this is indeed the case, shouldn't the respective acl category also be fast and not slow for us? After all those categories seem to have a purpose / reason for existence 
2. You can always add a mapping function with more complex conditions

We can keep it as is, just saying that a clean mapping is likely possible for almost all commands

Reply:
1. It could be that there is a mistake on our end as well. If the operation is not O(1) then it should not be `fast`

> We can keep it as is, just saying that a clean mapping is likely possible for almost all commands

I will take a look, if it can be done and save us a few lines then BIG YES. I will ping here

Reply:
@dranikpg inferring CAT::WRITE from CO::WRITE is meaningless if we plan to remove CO::WRITE in the future.

Reply:
In general, I agree with @kostasrim that any further changes are incremental, subjective and won't make a huge difference.

Reply:
Okay, not gonna interfere any further 😄

Reply:
However we'll still need some CO flags just for internal reasons (like no autojounal)